aioredis/connection.py

```diff
@@ -237,3 +237,16 @@ def auth(self, password):
     @asyncio.coroutine
     def get_atomic_connection(self):
         return self
+
+
+class RedisFuture(asyncio.Future):
+    def __init__(self, conn, *args, **kwargs):
+        super(RedisFuture, self).__init__(*args, **kwargs)
+        self._conn = conn
+
+    def cancel(self):
+        super(RedisFuture, self).cancel()
+        if self._conn._reader_task:
+            self._conn._reader_task.cancel()
+            self._conn._reader_task = asyncio.Task(self._conn._read_data(),
```

Comment:
this is bad, because cancelling `_reader_task` will also cancel `_reader.read()` inside it and make connection unusable. So re-createing new `_reader_task` does not make much sense.